assignments/closure.js

```diff
@@ -4,26 +4,66 @@
 // that manipulates variables defined in the outer scope.
 // The outer scope can be a parent function, or the top level of the script.
 
+function makeAdder(x) {
+  let Add = function(y) {
+    return x + y;
+  }
+  return Add;
+}
+
+var addTo = makeAdder(6);
+console.log(addTo(4));  // 7
 
 /* STRETCH PROBLEMS, Do not attempt until you have completed all previous tasks for today's project files */
 
 
 // ==== Challenge 2: Implement a "counter maker" function ====
+// IMPLEMENTATION OF counterMaker:
+// 1- Declare a `count` variable with a value of 0. We will be mutating it, so declare it using `let`!
+// 2- Declare a function `counter`. It should increment and return `count`.
+//      NOTE: This `counter` function, being nested inside `counterMaker`,
+//      "closes over" the `count` variable. It can "see" it in the parent scope!
+// 3- Return the `counter` function.
 const counterMaker = () => {
-  // IMPLEMENTATION OF counterMaker:
-  // 1- Declare a `count` variable with a value of 0. We will be mutating it, so declare it using `let`!
-  // 2- Declare a function `counter`. It should increment and return `count`.
-  //      NOTE: This `counter` function, being nested inside `counterMaker`,
-  //      "closes over" the `count` variable. It can "see" it in the parent scope!
-  // 3- Return the `counter` function.
+  let count = 0;
+  let counter = function(){
+    return count = count + 1;
+  }
+  return counter;
 };
+const myCounter = counterMaker();
+console.log(myCounter());
+console.log(myCounter());
+console.log(myCounter());
+console.log(myCounter());
+
+
 // Example usage: const myCounter = counterMaker();
 // myCounter(); // 1
 // myCounter(); // 2
 
 // ==== Challenge 3: Make `counterMaker` more sophisticated ====
 // It should have a `limit` parameter. Any counters we make with `counterMaker`
 // will refuse to go over the limit, and start back at 1.
+const counterMakers = () => {
```

Comment:
You aren't doing this programmatically. The counterMakers is supposed to take in an argument which is supposed to be the limit, and then set the limit of count based on that